homeassistant/components/smartenergy_awattar/sensor.py

```diff
@@ -0,0 +1,122 @@
+"""Platform for Awattar sensor integration."""
+
+from collections.abc import Callable, Mapping
+import logging
+from typing import Any
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import AWATTAR_COORDINATOR, DOMAIN, MANUFACTURER, UNIT
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class ForecastSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a sensor on the forecast of the energy prices integration."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        entity_id: str,
+    ) -> None:
+        """Initialize the Base sensor."""
+        super().__init__(coordinator)
+        self.entity_id: str = entity_id
+        self._name: str = "Awattar forecast"
+        self._unit: str = UNIT
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement of the sensor, if any."""
+        return self._unit
+
+    @property
+    def device_info(self) -> entity.DeviceInfo:
+        """Return the device information."""
+        return {
+            "identifiers": {(DOMAIN, self.entity_id)},
+            "name": self._name,
+            "manufacturer": MANUFACTURER,
+            "model": "",
+        }
+
+    @property
+    def capability_attributes(self) -> Mapping[str, Any] | None:
```

Comment:
Why are you setting capability attributes? I'm not sure what your data is, but sensors typically have a native_value that displays what they are

Reply:
I'm using attributes because the value is a forecast, meaning array of of values (in this case objects). That is not possible with native value as it doesn't work with such format and has size limitations. From what I've seen people either use attributes or have a sensor per array item which I don't really like from user point of view.